contracts/Verifiers/NftApprovalSwapVerifier.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.10;
+pragma abicoder v1;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "../Interfaces/ICallExecutor.sol";
+import "../Libraries/Bit.sol";
+import "../Libraries/NativeOrERC20.sol";
+
+/// @title Verifier for ERC721 swaps using ERC721.approve()
+/// @notice These functions should be executed by metaPartialSignedDelegateCall() on Brink account proxy contracts
+contract NftApprovalSwapVerifier {
+  using NativeOrERC20 for address;
+
+  ICallExecutor constant CALL_EXECUTOR = ICallExecutor(0xDE61dfE5fbF3F4Df70B16D0618f69B96A2754bf8);
+
+  /// @dev Verifies swap from fungible token (ERC20 or Native) to ERC721
+  /// @notice This should be executed by metaDelegateCall() or metaDelegateCall_EIP1271() with the following signed and unsigned params
+  /// @param bitmapIndex The index of the replay bit's bytes32 slot [signed]
+  /// @param bit The value of the replay bit [signed]
+  /// @param tokenIn The input token provided for the swap. Can be ERC20 or Native [signed]
+  /// @param nftOut The ERC721 output token required to be received from the swap [signed]
+  /// @param tokenInAmount Amount of tokenIn provided [signed]
+  /// @param expiryBlock The block when the swap expires [signed]
+  /// @param to Address of the contract that will fulfill the swap [unsigned]
+  /// @param data Data to execute on the `to` contract to fulfill the swap [unsigned]
+  function tokenToNft(
+    uint256 bitmapIndex, uint256 bit, address tokenIn, IERC721 nftOut, uint256 tokenInAmount, uint256 expiryBlock, address recipient,
+    address to, bytes calldata data
+  )
+    external
+  {
+    require(expiryBlock > block.number, 'Expired');
+  
+    Bit.useBit(bitmapIndex, bit);
+
+    address owner = proxyOwner();
+
+    uint256 nftOutBalance = nftOut.balanceOf(owner);
+
+    IERC20(tokenIn).transferFrom(owner, recipient, tokenInAmount);
+    CALL_EXECUTOR.proxyCall(to, data);
+
+    uint256 nftOutAmountReceived = nftOut.balanceOf(owner) - nftOutBalance;
```

Comment:
What if nftOut.balanceOf(owner) is somehow 0, but their initial balance is 1, would that break the following require check since the integer is unsigned?

Reply:
double checked this, solc 8 reverts on arithmetic overflows https://docs.soliditylang.org/en/v0.8.17/080-breaking-changes.html

Reply:
also this logic was part of the original audit of https://github.com/brinktrade/brink-verifiers/blob/master/contracts/Verifiers/LimitSwapVerifier.sol#L53)